cloudca/resource_cloudca_vpn.go

```diff
@@ -0,0 +1,147 @@
+package cloudca
+
+import (
+	"fmt"
+
+	"github.com/cloud-ca/go-cloudca"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudcaVpn() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudcaVpnCreate,
+		Read:   resourceCloudcaVpnRead,
+		Delete: resourceCloudcaVpnDelete,
+
+		Schema: map[string]*schema.Schema{
+			"environment_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the environment where the vpn should be created",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Id of the VPC for the vpn",
+			},
+			"certificate": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Certificate to use when using IKEV2 vpn type",
+			},
+			"preshared_key": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Preshared key to use when using L2TP vpn type",
+			},
+			"public_ip": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Public IP address associated with the vpn",
+			},
+			"public_ip_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "ID of the public IP address associated with the vpn",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "State of the vpn",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Type of vpn connection",
+			},
+		},
+	}
+}
+
+func resourceCloudcaVpnCreate(d *schema.ResourceData, meta interface{}) error {
+	vpnIPPurpose := "SOURCE_NAT"
+	ccaResources, rerr := getResourcesForEnvironmentID(meta.(*cca.CcaClient), d.Get("environment_id").(string))
+	if rerr != nil {
+		return rerr
+	}
+
+	var vpnPubIPID string
+	pubIps, _ := ccaResources.PublicIps.List()
+	for _, ip := range pubIps {
+		if ip.VpcId == d.Get("vpc_id").(string) {
+			for _, purpose := range ip.Purposes {
+				if purpose == vpnIPPurpose {
+					vpnPubIPID = ip.Id
+				}
+			}
+		}
+	}
+
+	if vpnPubIPID == "" {
+		return fmt.Errorf("Error enabling the VPN because no Source NAT IP was found for the VPC")
+	}
+
+	_, err := ccaResources.RemoteAccessVpn.Enable(vpnPubIPID)
```

Comment:
You need to catch the return value of `ccaResources.RemoteAccessVpn` and use it for `d.SetId()` below.

Reply:
No, the `Enable` is not a well formed ACS API, so it does not return the ID unfortunately.  It only returns `true`, so there is not reason to set the return to anything as the error is the only useful return this API call makes.

Reply:
I will look into tweaking the implementation in the `go-cloudca` lib to make it behave how we expect it to even though the underlying API does not. If I do this, we could use the returned value.

Reply:
Based on your other comments, that newly created VPN shares the ID of public IP, the above implementation is good. Thanks

Reply:
I am not sure if it is a good idea to make the `go-cloudca` library behave differently than the underlying API.  While it would be convenient to build some additional logic into the `go-cloudca` to make it behave like the other APIs (and return the ID as the other APIs do), I am not sure it is a good idea as we will be managing additional logic in the lib.  Thoughts?  @pdube & @khos2ow

Reply:
I think we should be keeping the library as minimalist as possible

Reply:
I agree, we should keep `go-cloudca` as close to the actual underlying API and if it requires adjustment that can be done on the program which uses the library (e.g. this terraform provider)